packages/in_app_purchase/in_app_purchase_storekit/lib/src/in_app_purchase_storekit_platform.dart

```diff
@@ -81,6 +81,20 @@ class InAppPurchaseStoreKitPlatform extends InAppPurchasePlatform {
     return true; // There's no error feedback from iOS here to return.
   }
 
+  static Future<bool> buyNonConsumableWithDiscount(
+      {required PurchaseParam purchaseParam, SKPaymentDiscountWrapper? discount}) async {
+    await _skPaymentQueueWrapper.addPayment(SKPaymentWrapper(
```

Comment:
Why is it a static method while it is using an instance variable?

Reply:
I would imagine the discount can be part of the `PurchaseParam`

Reply:
@cyanglaz the instance variable is also static. I agree it is not the best option. I merely wanted a working solution to open a pull request with. Then have a discussion about how to resolve it, or have it resolved by someone from the maintainers.

If it's part of PurchaseParam, this method would not be needed at all but I presume there should be also Android counterpart . In the original Pull request I saw only iOS implementation. I don't know how the Android offers work (if there are any).